Bug: fs/promises breaks ESM usage in Virtual Screen Reader

# Bug: fs/promises breaks ESM usage (runVbsScript)

Setup:

- Node version: 20
- Guidepup version: 0.22
- OS Platform: mac
- OS Release: sonoma

## Details

When trying to use VSR in a Storybook story I got the following error when importing `virtual` via `import { virtual } from '@guidepup/virtual-screen-reader';`. I only had the import statement and did not implement `virtual` in the file.

```
✘ [ERROR] Could not read from file: /Users/scottnath/development/profile-components/node_modules/node-stdlib-browser/cjs/mock/empty.js/promises

    node_modules/@guidepup/guidepup/lib/windows/runVbsScript.js:5:27:
      5 │ const promises_1 = require("node:fs/promises");
        ╵                            ~~~~~~~~~~~~~~~~~~

Unhandled promise rejection: Error: Build failed with 1 error:
node_modules/@guidepup/guidepup/lib/windows/runVbsScript.js:5:27: ERROR: Could not read from file: /Users/scottnath/development/profile-components/node_modules/node-stdlib-browser/cjs/mock/empty.js/promises
```

Searching led me to [this issue in virtual-screen-reader](https://github.com/guidepup/virtual-screen-reader/issues/49). After a lot of experiments, I found that the usage of `fs/promises' for an import in `lib/windows/runVbsScript.js` was causing the error. If I change that file to be:

```
import { join, sep } from "path";
import { promises } from "node:fs";
import { execFile } from "child_process";
import { tmpdir } from "os";

const { mkdtemp, realpath, rm, writeFile } = promises;
```

then the `compile` step changes the output from

```
const promises_1 = require("node:fs/promises");
const child_process_1 = require("child_process");
const os_1 = require("os");
```

to:

```
const node_fs_1 = require("node:fs");
const child_process_1 = require("child_process");
const os_1 = require("os");
const { mkdtemp, realpath, rm, writeFile } = node_fs_1.promises;
```

This change stops the error when used by VSR in my stories files.

--- a/src/windows/runVbsScript.test.ts
+++ b/src/windows/runVbsScript.test.ts
@@ -1,6 +1,6 @@
 import { ChildProcess, ExecException, execFile } from "child_process";
-import { mkdtemp, realpath, rm, writeFile } from "fs/promises";
 import { DEFAULT_MAX_BUFFER } from "../constants";
+import { promises } from "fs";
 import { runVbsScript } from "./runVbsScript";
 import { sep } from "path";
 import { tmpdir } from "os";
@@ -9,11 +9,13 @@ jest.mock("child_process", () => ({
   execFile: jest.fn(),
 }));
 
-jest.mock("fs/promises", () => ({
-  mkdtemp: jest.fn(),
-  realpath: jest.fn(),
-  rm: jest.fn(),
-  writeFile: jest.fn(),
+jest.mock("fs", () => ({
+  promises: {
+    mkdtemp: jest.fn(),
+    realpath: jest.fn(),
+    rm: jest.fn(),
+    writeFile: jest.fn(),
+  },
 }));
 
 const script = "test-script";
@@ -28,10 +30,10 @@ const dirStub = "test-dir";
 const tmpdirStub = "test-tmpdir";
 
 const mockExecFile = jest.mocked(execFile);
-const mockMkdtemp = jest.mocked(mkdtemp);
-const mockRealpath = jest.mocked(realpath);
-const mockRm = jest.mocked(rm);
-const mockWriteFile = jest.mocked(writeFile);
+const mockMkdtemp = jest.mocked(promises.mkdtemp);
+const mockRealpath = jest.mocked(promises.realpath);
+const mockRm = jest.mocked(promises.rm);
+const mockWriteFile = jest.mocked(promises.writeFile);
 
 describe("runVbsScript", () => {
   let resultPromise: Promise<string | void>;

--- a/src/windows/runVbsScript.ts
+++ b/src/windows/runVbsScript.ts
@@ -1,7 +1,9 @@
 import { join, sep } from "path";
-import { mkdtemp, realpath, rm, writeFile } from "fs/promises";
 import { execFile } from "child_process";
+import { promises } from "fs";
 import { tmpdir } from "os";
+
+const { mkdtemp, realpath, rm, writeFile } = promises;
 
 const DEFAULT_MAX_BUFFER = 1000 * 1000 * 100;
 const CSCRIPT = "cscript";

